FullStackMVCAPP/FullStackMVCAPP/DataContext/Repositories/CharacterRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using FullStackMVCAPP.Repositories;
+using FullStackMVCAPP.Repositories.Base;
+
+namespace FullStackMVCAPP.DataContext.Repositories
+{
+    public class CharacterRepository: IEntityRepository<FullStackMVCAPP.Models.Character>, ICharacterRepositry
+    {
+
+
+        public IList<FullStackMVCAPP.Models.Character> EntityList(GOTContext _GOTContext)
+        {
+            return _GOTContext.Characters.ToList();
+        }
+
+        public FullStackMVCAPP.Models.Character GetEntityByID(GOTContext _GOTContext, int id)
+        {
+            return _GOTContext.Characters.Where(chr => chr.Id == id).First();
+        }
+
+        public IList<FullStackMVCAPP.Models.Character> GetCharacterByHouseId(GOTContext _GOTContext, int id)
+        {
+            var house = _GOTContext.Houses.Where(hos => hos.Id == id).First();
+            return _GOTContext.Characters.Where(chr => chr.HouseID.Id == house.Id).ToList();
+        }
```

Comment:
This query could be greatly simplified as there are a lot of circular referencing here. As the `id` parameter is the `HouseId` passed from the view, it would be easier to refactor this:
```suggestion
        public List<Character> GetCharactersByHouseId(GOTContext _GOTContext, int houseId)
        {
            return _GOTContext.Characters.Where(chr => chr.HouseID.Id == houseId).ToList();
        }
```

Reply:
I need the house object for the cshtml to display the house name, you can't use character last name due to some of them having "bastard" names such as snow, flowers, stome, etc, however I will fix the bottom line to use the parameter int id

Reply:
Under normal circumstanes you would have a virtual property of the House class that would return the House that each character is a part of. Once you return the character, the House property would be accessible. Can give a demo later if that would help?